tests/EvmLookupServer/Test_EvmLookupServer.cpp

```diff
@@ -1007,19 +1014,24 @@ BOOST_AUTO_TEST_CASE(test_eth_get_transaction_by_hash) {
   // txBlock)
   std::vector<TransactionWithReceipt> transactions;
 
+  constexpr uint64_t EPOCH_NUM = 1;
   constexpr uint32_t TRANSACTIONS_COUNT = 2;
+
   for (uint32_t i = 0; i < TRANSACTIONS_COUNT; ++i) {
-    transactions.emplace_back(constructTxWithReceipt(i, pairOfKey));
+    transactions.emplace_back(constructTxWithReceipt(i, pairOfKey, EPOCH_NUM));
   }
 
-  constexpr uint64_t EPOCH_NUM = 1;
   for (const auto& transaction : transactions) {
     bytes body;
     transaction.Serialize(body, 0);
     BlockStorage::GetBlockStorage().PutTxBody(
         EPOCH_NUM, transaction.GetTransaction().GetTranID(), body);
   }
 
+  // Need to create a block with our TXs in since that is referenced in the
+  // transaction receipt (tx index)
+  buildCommonEthBlockCase(mediator, EPOCH_NUM, transactions, pairOfKey);
```

Comment:
Similarly, there needs to be a block matching the block number that the TX receipt is of. This is used when getting TX by hash etc since the TX receipt needs to know its index in the block etc.